lib/cucumber/glue/proto_world.rb

```diff
@@ -108,8 +108,7 @@ def embed(file, mime_type, _label = 'Screenshot')
       end
 
       def log(message)
-        raise Cucumber::LogTypeInvalid unless message.is_a?(String)
-        attach(message.dup, 'text/x.cucumber.log+plain')
+        attach(message.to_s.dup, 'text/x.cucumber.log+plain')
```

Comment:
Why the `dup`?

Reply:
Otherwise we log the object itself and it will display its value at display time, not at log time (there a test showing that and it fails without the `dup`)

Reply:
I understand why we used to do `message.dup`, but I don't think we need `message.to_s.dup`. I think `message.to_s` is sufficient.

Reply:
(Me, a reader, _thought_ this was because the string would be used in a mutable context.)

Reply:
I though the `to_s` would be enough too. In fact it's enough for non-string objects (obviously :D ) but when using a String we need the `dup`.

An example from the tests:

```ruby
        describe 'when modifying the printed variable after the call to puts' do
          define_feature <<-FEATURE
        Feature: Banana party

          Scenario: Monkey eats banana
            When puts is called twice for the same variable
          FEATURE

          define_steps do
            When(/^puts is called twice for the same variable$/) do
              foo = String.new('a')
              puts foo
              foo.upcase!
              puts foo
            end
          end

```

This test fails without the `dup`. I suspect that String.to_s do not return a duplicate but imply itself.

And @olleolleolle you're not wrong, it's just that the string might be mutated between the moment it I logged and the moment we actually display it.